test/functional/plugin/lsp_spec.lua

```diff
@@ -2388,9 +2386,7 @@ describe('LSP', function()
     it(
       'Supports file creation in folder that needs to be created with CreateFile payload',
       function()
-        local tmpfile = tmpname()
-        os.remove(tmpfile) -- Should not exist, only interested in a tmpname
-        tmpfile = tmpfile .. '/dummy/x/'
+        local tmpfile = tmpname(false) .. '/dummy/x/'
```

Comment:
Looks like many tests need this functionality so I added a `create:boolean` param to `t.tmpname()`. Will followup in another PR to improve it.